google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldPath.java

```diff
@@ -32,6 +32,7 @@
 public abstract class FieldPath extends BasePath<FieldPath> implements Comparable<FieldPath> {
 
   private static final String DOCUMENT_ID_SENTINEL = "__name__";
+  public static boolean splitOnDots = false;
```

Comment:
you don't need this new field, or it can be private and non static

Reply:
Without making static i can not access it in ```fromDotSeparatedString(String field)``` method.

Reply:
static property belongs to ALL instances of the class. In your case you want to set it for a single object.
I think you don't need this field at all